rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -295,24 +305,119 @@ namespace rsx
 						// Actually uses the modulo operator
 						_min_index = 0;
 						_max_index = attrib.frequency - 1;
+
+						if (freq_count != umax)
+						{
+							if (std::find(std::begin(frequencies), std::begin(frequencies) + freq_count, attrib.frequency) == std::begin(frequencies) + freq_count)
+							{
+								frequencies[freq_count++] = attrib.frequency;
+							}
+						}
 					}
 					else
 					{
 						// Same as having no modulo
+						freq_count = umax;
 						_max_index = max_index;
 					}
 				}
 				else
 				{
 					// Division operator
+					freq_count = umax;
 					_min_index = std::min(_min_index, first / attrib.frequency);
 					_max_index = std::max<u32>(_max_index, utils::aligned_div(max_index, attrib.frequency));
 				}
 			}
 		}
 
+		while (freq_count > 0 && freq_count != umax)
```

Comment:
what's the purpose of a loop if you always break in the first iteration?

Reply:
because `break`.

Reply:
makes no sense.

Reply:
But this is C++, you can't break inside a regular block.